build warning on macOS

```shell
cc -I/Users/zhaozg/iOS/include/luajit-2.1 -DCURL_STATICLIB -I/Users/zhaozg/iOS/include  -O2 -g -pipe -fPIC -undefined dynamic_lookup  -Wall -Wno-unused-value -DPTHREADS  -c -o src/l52util.o src/l52util.c
clang: warning: argument unused during compilation: '-undefined dynamic_lookup' [-Wunused-command-line-argument]
cc -I/Users/zhaozg/iOS/include/luajit-2.1 -DCURL_STATICLIB -I/Users/zhaozg/iOS/include  -O2 -g -pipe -fPIC -undefined dynamic_lookup  -Wall -Wno-unused-value -DPTHREADS  -c -o src/lchttppost.o src/lchttppost.c
clang: warning: argument unused during compilation: '-undefined dynamic_lookup' [-Wunused-command-line-argument]
src/lchttppost.c:119:12: warning: unused function 'lcurl_hpost_to_s' [-Wunused-function]
static int lcurl_hpost_to_s(lua_State *L){
           ^
1 warning generated.
cc -I/Users/zhaozg/iOS/include/luajit-2.1 -DCURL_STATICLIB -I/Users/zhaozg/iOS/include  -O2 -g -pipe -fPIC -undefined dynamic_lookup  -Wall -Wno-unused-value -DPTHREADS  -c -o src/lcurl.o src/lcurl.c
clang: warning: argument unused during compilation: '-undefined dynamic_lookup' [-Wunused-command-line-argument]
cc -I/Users/zhaozg/iOS/include/luajit-2.1 -DCURL_STATICLIB -I/Users/zhaozg/iOS/include  -O2 -g -pipe -fPIC -undefined dynamic_lookup  -Wall -Wno-unused-value -DPTHREADS  -c -o src/lceasy.o src/lceasy.c
clang: warning: argument unused during compilation: '-undefined dynamic_lookup' [-Wunused-command-line-argument]
src/lceasy.c:118:15: warning: unused variable 'code' [-Wunused-variable]
    CURLMcode code = lcurl__multi_remove_handle(L, p->multi, p);
              ^
src/lceasy.c:1703:3: warning: unused typedef 'static_assert_failed_at_line_1703' [-Wunused-local-typedef]
  LCURL_ASSERT_SAME_OFFSET(lcurl_easy_t, L,       lcurl_hpost_stream_t, L);
  ^
src/lcutils.h:36:48: note: expanded from macro 'LCURL_ASSERT_SAME_OFFSET'
#define LCURL_ASSERT_SAME_OFFSET(a, am, b, bm) LCURL_STATIC_ASSERT( (offsetof(a,am)) == (offsetof(b,bm)) )
                                               ^
src/lcutils.h:33:48: note: expanded from macro 'LCURL_STATIC_ASSERT'
#define LCURL_STATIC_ASSERT(expr) typedef char LCURL_CONCAT_STATIC_ASSERT_IMPL_(static_assert_failed_at...
                                               ^
src/lcutils.h:31:48: note: expanded from macro 'LCURL_CONCAT_STATIC_ASSERT_IMPL_'
#define LCURL_CONCAT_STATIC_ASSERT_IMPL_(x, y) LCURL_CONCAT1_STATIC_ASSERT_IMPL_ (x, y)
                                               ^
src/lcutils.h:32:49: note: expanded from macro 'LCURL_CONCAT1_STATIC_ASSERT_IMPL_'
#define LCURL_CONCAT1_STATIC_ASSERT_IMPL_(x, y) x##y
                                                ^
<scratch space>:158:1: note: expanded from here
static_assert_failed_at_line_1703
^
src/lceasy.c:1702:3: warning: unused typedef 'static_assert_failed_at_line_1702' [-Wunused-local-typedef]
  LCURL_ASSERT_SAME_OFFSET(lcurl_easy_t, magic,   lcurl_hpost_stream_t, magic);
  ^
src/lcutils.h:36:48: note: expanded from macro 'LCURL_ASSERT_SAME_OFFSET'
#define LCURL_ASSERT_SAME_OFFSET(a, am, b, bm) LCURL_STATIC_ASSERT( (offsetof(a,am)) == (offsetof(b,bm)) )
                                               ^
src/lcutils.h:33:48: note: expanded from macro 'LCURL_STATIC_ASSERT'
#define LCURL_STATIC_ASSERT(expr) typedef char LCURL_CONCAT_STATIC_ASSERT_IMPL_(static_assert_failed_at...
                                               ^
src/lcutils.h:31:48: note: expanded from macro 'LCURL_CONCAT_STATIC_ASSERT_IMPL_'
#define LCURL_CONCAT_STATIC_ASSERT_IMPL_(x, y) LCURL_CONCAT1_STATIC_ASSERT_IMPL_ (x, y)
                                               ^
src/lcutils.h:32:49: note: expanded from macro 'LCURL_CONCAT1_STATIC_ASSERT_IMPL_'
#define LCURL_CONCAT1_STATIC_ASSERT_IMPL_(x, y) x##y
                                                ^
<scratch space>:156:1: note: expanded from here
static_assert_failed_at_line_1702
^
src/lceasy.c:1706:3: warning: unused typedef 'static_assert_failed_at_line_1706' [-Wunused-local-typedef]
  LCURL_ASSERT_SAME_FIELD_SIZE(lcurl_easy_t, rbuffer, lcurl_hpost_stream_t, rbuffer);
  ^
src/lcutils.h:37:52: note: expanded from macro 'LCURL_ASSERT_SAME_FIELD_SIZE'
#define LCURL_ASSERT_SAME_FIELD_SIZE(a, am, b, bm) LCURL_ASSERT_SAME_SIZE(((a*)0)->am, ((b*)0)->bm)
                                                   ^
src/lcutils.h:35:38: note: expanded from macro 'LCURL_ASSERT_SAME_SIZE'
#define LCURL_ASSERT_SAME_SIZE(a, b) LCURL_STATIC_ASSERT( sizeof(a) == sizeof(b) )
                                     ^
src/lcutils.h:33:48: note: expanded from macro 'LCURL_STATIC_ASSERT'
#define LCURL_STATIC_ASSERT(expr) typedef char LCURL_CONCAT_STATIC_ASSERT_IMPL_(static_assert_failed_at...
                                               ^
src/lcutils.h:31:48: note: expanded from macro 'LCURL_CONCAT_STATIC_ASSERT_IMPL_'
#define LCURL_CONCAT_STATIC_ASSERT_IMPL_(x, y) LCURL_CONCAT1_STATIC_ASSERT_IMPL_ (x, y)
                                               ^
src/lcutils.h:32:49: note: expanded from macro 'LCURL_CONCAT1_STATIC_ASSERT_IMPL_'
#define LCURL_CONCAT1_STATIC_ASSERT_IMPL_(x, y) x##y
                                                ^
<scratch space>:164:1: note: expanded from here
static_assert_failed_at_line_1706
^
src/lceasy.c:1704:3: warning: unused typedef 'static_assert_failed_at_line_1704' [-Wunused-local-typedef]
  LCURL_ASSERT_SAME_OFFSET(lcurl_easy_t, rd,      lcurl_hpost_stream_t, rd);
  ^
src/lcutils.h:36:48: note: expanded from macro 'LCURL_ASSERT_SAME_OFFSET'
#define LCURL_ASSERT_SAME_OFFSET(a, am, b, bm) LCURL_STATIC_ASSERT( (offsetof(a,am)) == (offsetof(b,bm)) )
                                               ^
src/lcutils.h:33:48: note: expanded from macro 'LCURL_STATIC_ASSERT'
#define LCURL_STATIC_ASSERT(expr) typedef char LCURL_CONCAT_STATIC_ASSERT_IMPL_(static_assert_failed_at...
                                               ^
src/lcutils.h:31:48: note: expanded from macro 'LCURL_CONCAT_STATIC_ASSERT_IMPL_'
#define LCURL_CONCAT_STATIC_ASSERT_IMPL_(x, y) LCURL_CONCAT1_STATIC_ASSERT_IMPL_ (x, y)
                                               ^
src/lcutils.h:32:49: note: expanded from macro 'LCURL_CONCAT1_STATIC_ASSERT_IMPL_'
#define LCURL_CONCAT1_STATIC_ASSERT_IMPL_(x, y) x##y
                                                ^
<scratch space>:160:1: note: expanded from here
static_assert_failed_at_line_1704
^
src/lceasy.c:1705:3: warning: unused typedef 'static_assert_failed_at_line_1705' [-Wunused-local-typedef]
  LCURL_ASSERT_SAME_OFFSET(lcurl_easy_t, rbuffer, lcurl_hpost_stream_t, rbuffer);
  ^
src/lcutils.h:36:48: note: expanded from macro 'LCURL_ASSERT_SAME_OFFSET'
#define LCURL_ASSERT_SAME_OFFSET(a, am, b, bm) LCURL_STATIC_ASSERT( (offsetof(a,am)) == (offsetof(b,bm)) )
                                               ^
src/lcutils.h:33:48: note: expanded from macro 'LCURL_STATIC_ASSERT'
#define LCURL_STATIC_ASSERT(expr) typedef char LCURL_CONCAT_STATIC_ASSERT_IMPL_(static_assert_failed_at...
                                               ^
src/lcutils.h:31:48: note: expanded from macro 'LCURL_CONCAT_STATIC_ASSERT_IMPL_'
#define LCURL_CONCAT_STATIC_ASSERT_IMPL_(x, y) LCURL_CONCAT1_STATIC_ASSERT_IMPL_ (x, y)
                                               ^
src/lcutils.h:32:49: note: expanded from macro 'LCURL_CONCAT1_STATIC_ASSERT_IMPL_'
#define LCURL_CONCAT1_STATIC_ASSERT_IMPL_(x, y) x##y
                                                ^
<scratch space>:162:1: note: expanded from here
static_assert_failed_at_line_1705
^
6 warnings generated.
cc -I/Users/zhaozg/iOS/include/luajit-2.1 -DCURL_STATICLIB -I/Users/zhaozg/iOS/include  -O2 -g -pipe -fPIC -undefined dynamic_lookup  -Wall -Wno-unused-value -DPTHREADS  -c -o src/lcmulti.o src/lcmulti.c
clang: warning: argument unused during compilation: '-undefined dynamic_lookup' [-Wunused-command-line-argument]
cc -I/Users/zhaozg/iOS/include/luajit-2.1 -DCURL_STATICLIB -I/Users/zhaozg/iOS/include  -O2 -g -pipe -fPIC -undefined dynamic_lookup  -Wall -Wno-unused-value -DPTHREADS  -c -o src/lcutils.o src/lcutils.c
clang: warning: argument unused during compilation: '-undefined dynamic_lookup' [-Wunused-command-line-argument]
cc -I/Users/zhaozg/iOS/include/luajit-2.1 -DCURL_STATICLIB -I/Users/zhaozg/iOS/include  -O2 -g -pipe -fPIC -undefined dynamic_lookup  -Wall -Wno-unused-value -DPTHREADS  -c -o src/lcerror.o src/lcerror.c
clang: warning: argument unused during compilation: '-undefined dynamic_lookup' [-Wunused-command-line-argument]
cc -I/Users/zhaozg/iOS/include/luajit-2.1 -DCURL_STATICLIB -I/Users/zhaozg/iOS/include  -O2 -g -pipe -fPIC -undefined dynamic_lookup  -Wall -Wno-unused-value -DPTHREADS  -c -o src/lcshare.o src/lcshare.c
clang: warning: argument unused during compilation: '-undefined dynamic_lookup' [-Wunused-command-line-argument]
cc -shared -I/Users/zhaozg/iOS/include/luajit-2.1 -DCURL_STATICLIB -I/Users/zhaozg/iOS/include  -O2 -g -pipe -fPIC -undefined dynamic_lookup  -Wall -Wno-unused-value -DPTHREADS  -o lcurl.so src/l52util.o src/lchttppost.o src/lcurl.o src/lceasy.o src/lcmulti.o src/lcutils.o src/lcerror.o src/lcshare.o -ldl -L/Users/zhaozg/iOS/lib -l
```

machine info
```shell
MacBook-Air:Lua-cURLv3 zhaozg$ uname -a
Darwin MacBook-Air 16.7.0 Darwin Kernel Version 16.7.0: Thu Jun 15 17:36:27 PDT 2017; root:xnu-3789.70.16~2/RELEASE_X86_64 x86_64
```

--- a/src/lceasy.c
+++ b/src/lceasy.c
@@ -124,7 +124,7 @@ static int lcurl_easy_cleanup(lua_State *L){
   int i;
 
   if(p->multi){
-    CURLMcode code = lcurl__multi_remove_handle(L, p->multi, p);
+    LCURL_UNUSED_VAR CURLMcode code = lcurl__multi_remove_handle(L, p->multi, p);
 
     //! @todo what I can do if I can not remove it???
   }

--- a/src/lcutils.h
+++ b/src/lcutils.h
@@ -25,11 +25,19 @@
 #  define LCURL_CC_SUPPORT_FORWARD_TYPEDEF 0
 #endif
 
+#ifdef __GNUC__
+  #define LCURL_UNUSED_TYPEDEF __attribute__ ((unused))
+#else
+  #define LCURL_UNUSED_TYPEDEF
+#endif
+
+#define LCURL_UNUSED_VAR LCURL_UNUSED_TYPEDEF
+
 #define LCURL_MAKE_VERSION(MIN, MAJ, PAT) ((MIN<<16) + (MAJ<<8) + PAT)
 #define LCURL_CURL_VER_GE(MIN, MAJ, PAT) (LIBCURL_VERSION_NUM >= LCURL_MAKE_VERSION(MIN, MAJ, PAT))
 
 #define LCURL_CONCAT_STATIC_ASSERT_IMPL_(x, y) LCURL_CONCAT1_STATIC_ASSERT_IMPL_ (x, y)
-#define LCURL_CONCAT1_STATIC_ASSERT_IMPL_(x, y) x##y
+#define LCURL_CONCAT1_STATIC_ASSERT_IMPL_(x, y) LCURL_UNUSED_TYPEDEF x##y
 #define LCURL_STATIC_ASSERT(expr) typedef char LCURL_CONCAT_STATIC_ASSERT_IMPL_(static_assert_failed_at_line_, __LINE__) [(expr) ? 1 : -1]
 
 #define LCURL_ASSERT_SAME_SIZE(a, b) LCURL_STATIC_ASSERT( sizeof(a) == sizeof(b) )

